yescrypt/tests/kats.rs

```diff
@@ -0,0 +1,251 @@
+//! Adapted from the reference implementation's `TESTS-OK`.
+//! <https://github.com/openwall/yescrypt/blob/e5873f8/TESTS-OK>
+
+use hex_literal::hex;
+use std::{mem, ptr};
+use yescrypt::{yescrypt_init_local, yescrypt_kdf, Flags, Local, Params};
+
+// Test function whose parameters match the upstream KATs
+fn yescrypt(
+    passwd: &[u8],
+    salt: &[u8],
+    flags: Flags,
+    n: u64,
+    r: u32,
+    p: u32,
+    t: u32,
+    g: u32,
+    dstlen: usize,
+) -> Vec<u8> {
+    let params = Params {
+        flags,
+        N: n,
+        r,
+        p,
+        t,
+        g,
+        NROM: 0,
+    };
+
+    let mut local: Local = unsafe { mem::zeroed() };
+    unsafe {
+        yescrypt_init_local(&mut local);
+    }
+
+    let mut dst = vec![0u8; dstlen];
+
+    unsafe {
+        yescrypt_kdf(
+            ptr::null(),
+            &mut local,
+            passwd.as_ptr(),
+            passwd.len() as u64,
+            salt.as_ptr(),
+            salt.len() as u64,
+            &params,
+            dst.as_mut_ptr(),
+            dstlen as u64,
+        )
+    };
+    dst
+}
+
+// yescrypt("", "", 0, 16, 1, 1, 0, 0) = 77 d6 57 62 38 65 7b 20 3b 19 ca 42 c1 8a 04 97 f1 6b 48 44 e3 07 4a e8 df df fa 3f ed e2 14 42 fc d0 06 9d ed 09 48 f8 32 6a 75 3a 0f c8 1f 17 e8 d3 e0 fb 2e 0d 36 28 cf 35 e2 0c 38 d1 89 06
```

Comment:
Maybe we can use link to source of this line instead?

Reply:
You want me to add a line number to the link at the top of the file? (this particular location is now outdated)